src/devices/StUsb4500/samples/StUsb4500.Sample.cs

```diff
@@ -0,0 +1,180 @@
+        private static void UpdateNvmData(StUsb4500 stUsb)
+        {
+            byte[] nvmData =
+                {
+                    0x00, 0x00, 0xB0, 0xAA, 0x00, 0x45, 0x00, 0x00,
+                    0x10, 0x40, 0x9C, 0x1C, 0xF0, 0x01, 0x00, 0xDF,
+                    0x02, 0x40, 0x0F, 0x00, 0x32, 0x00, 0xFC, 0xF1,
+                    0x00, 0x19, 0x57, 0xAF, 0xF6, 0x4F, 0xFF, 0x00,
+                    0x00, 0x4B, 0x90, 0x21, 0x03, 0x00, 0x48, 0xFB,
+                };
+            stUsb.NvmData = nvmData;
+            PrintNvmData(stUsb);
+        }
+
+        /// <summary>Prints the source PDOs.</summary>
+        /// <param name="stUsb">The device.</param>
+        private static void PrintSourcePdo(StUsb4500 stUsb)
+        {
+            Power maxPower = Power.Zero;
+            Console.WriteLine("Source PDOs:");
+            PowerDeliveryObject[] sourcePdos = stUsb.SourcePowerDeliveryObjects;
+            for (int i = 0; i < sourcePdos.Length; i++)
+            {
+                PowerDeliveryObject pdo = sourcePdos[i];
+                Console.WriteLine($"PDO #{i + 1}: {pdo}");
+                if (pdo.Power > maxPower)
+                {
+                    maxPower = pdo.Power;
+                }
+            }
+
+            Console.WriteLine($"P(max) = {maxPower:0.##}{Environment.NewLine}");
+        }
+
+        /// <summary>Prints the RDO.</summary>
+        /// <param name="stUsb">The device.</param>
+        private static void PrintRdo(StUsb4500 stUsb)
+        {
+            Console.WriteLine("RDO (negotiated power):");
+            RequestDataObject rdo = stUsb.RequestDataObject;
+            Console.WriteLine($"Requested position: PDO #{rdo.ObjectPosition}");
+            Console.WriteLine($"Operating current: {rdo.OperatingCurrent:0.##}");
+            Console.WriteLine($"Max. current: {rdo.MaximalCurrent:0.##}");
+            Console.WriteLine($"USB communications capable: {rdo.UsbCommunicationsCapable}");
+            Console.WriteLine($"Capability mismatch: {rdo.CapabilityMismatch}");
+            ElectricPotentialDc voltage = stUsb.RequestedVoltage;
+            Console.WriteLine($"Requested voltage: {voltage:0.##}");
+            Console.WriteLine($"Available Power: {Power.FromWatts(voltage.VoltsDc * rdo.MaximalCurrent.Amperes):0.##}{Environment.NewLine}");
+        }
+
+        /// <summary>Updates the local PDOs.</summary>
+        /// <param name="stUsb">The device.</param>
+        private static void UpdateLocalPdo(StUsb4500 stUsb)
+        {
+            Console.WriteLine($"Update Local sink PDOs...{Environment.NewLine}");
+            PowerDeliveryObject[] sinkPdos = stUsb.SinkPowerDeliveryObjects;
+            if (sinkPdos.Length > 1 && sinkPdos[1] is FixedSupplyObject fixedSupplyObject1)
+            {
+                fixedSupplyObject1.Voltage = ElectricPotentialDc.FromVoltsDc(12);
+                fixedSupplyObject1.OperationalCurrent = ElectricCurrent.FromAmperes(2.0);
```

Comment:
if OperationalCurrent is available on all or most of the implementations it should be on the base class to avoid having to repeat the code like here. If it's not available on some of the implementations you should just add property: SupportsOperationallCurrent and check for it in here. Only when it's available on a single implementation is when it should not be on the base class

Reply:
(similar comment for other properties)

Reply:
@krwq This property is available in 2 of 3 possibilities. And 2 other properties are available in other 2 of the three. I'm not sure what way around is better, but to avoid mistakingly choosing the wrong one I tend to the current solution. Or should I change it to the suggested `Supports*` approach and throw `NotSupportedException` if the wrong one is used?

Reply:
fine as is